ocaml/xapi/xapi_vm.ml

```diff
@@ -1638,3 +1638,48 @@ let set_uefi_mode ~__context ~self ~mode =
   let id = Db.VM.get_uuid ~__context ~self in
   let args = [id; Helpers.uefi_mode_to_string mode] in
   Helpers.call_script !Xapi_globs.varstore_sb_state args
+
+let get_secureboot_readiness ~__context ~self =
+  let vmr = Db.VM.get_record ~__context ~self in
+  match Xapi_xenops.firmware_of_vm vmr with
+  | Bios ->
+      (* VM is not UEFI *)
+      `not_supported
+  | Uefi _ -> (
+      let platformdata = Db.VM.get_platform ~__context ~self in
+      match
+        Vm_platform.is_true ~key:"secureboot" ~platformdata ~default:false
+      with
+      | false ->
+          `disabled (* Secure boot is disabled *)
+      | true -> (
+        (* Secureboot is enabled *)
+        match
+          List.assoc_opt "EFI-variables" (Db.VM.get_NVRAM ~__context ~self)
+        with
+        | None ->
+            `first_boot (* TO BE VERIFIED *)
```

Comment:
Is there a plan to verify this?

Reply:
It's been tested and verified, I'll retest it after the latest changes and remove the comment after revalidation.

Reply:
The commit needs to be removed, otherwise it's good to go